feature-libs/product-configurator/rulebased/components/attribute/types/multi-selection-bundle/configurator-attribute-multi-selection-bundle.component.ts

```diff
@@ -0,0 +1,290 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Configurator } from '../../../../core/model/configurator.model';
+import { ConfigFormUpdateEvent } from '../../../form/configurator-form.event';
+import { ConfiguratorPriceComponentOptions } from '../../../price/configurator-price.component';
+import { ConfiguratorAttributeProductCardComponentOptions } from '../../product-card/configurator-attribute-product-card.component';
+import {
+  ConfiguratorAttributeQuantityComponentOptions,
+  Quantity,
+} from '../../quantity/configurator-attribute-quantity.component';
+import { ConfiguratorAttributeQuantityService } from '../../quantity/configurator-attribute-quantity.service';
+import { ConfiguratorAttributeBaseComponent } from '../base/configurator-attribute-base.component';
+
+interface SelectionValue {
+  name: string | undefined;
+  quantity: number | undefined;
+  selected: boolean | undefined;
+  valueCode: string | undefined;
+}
+
+@Component({
+  selector: 'cx-configurator-attribute-multi-selection-bundle',
+  templateUrl: './configurator-attribute-multi-selection-bundle.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ConfiguratorAttributeMultiSelectionBundleComponent
+  extends ConfiguratorAttributeBaseComponent
+  implements OnInit {
+  preventAction$ = new BehaviorSubject<boolean>(false);
+  multipleSelectionValues: SelectionValue[] = [];
+  loading$ = new BehaviorSubject<boolean>(false);
+
+  @Input() attribute: Configurator.Attribute;
+  @Input() ownerKey: string;
+
+  @Output() selectionChange = new EventEmitter<ConfigFormUpdateEvent>();
+
+  constructor(protected quantityService: ConfiguratorAttributeQuantityService) {
+    super();
+  }
+
+  ngOnInit() {
+    if (this.attribute.values && this.attribute.values.length > 0) {
+      this.multipleSelectionValues = this.attribute.values.map(
+        ({ name, quantity, selected, valueCode }) => ({
+          name,
+          quantity,
+          selected,
+          valueCode,
+        })
+      );
+    }
+
+    if (
+      this.attribute?.required &&
+      this.multipleSelectionValues.filter((value) => value.selected).length < 2
+    ) {
+      this.preventAction$.next(true);
+    }
+  }
+
+  /**
+   * Returns true if the quantity control should be displayed on attribute level for this component.
+   *
+   * @return {boolean} - Returns true if quantity control should be displayed.
+   */
+  get withQuantityOnAttributeLevel(): boolean {
```

Comment:
This is also in `ConfiguratorAttributeCheckBoxListComponent`, can we move it to parent component (or to some service)?

Reply:
Picked

Reply:
Done. The common logic is moved to the parent component, the quantity related logic - into the quantity service class.